src/geometryRTheta/time_solver/bsl_predcorr.hpp

```diff
@@ -123,74 +123,87 @@ class BslPredCorrRTheta : public ITimeSolverRTheta
 
 
         // Grid. ------------------------------------------------------------------------------------------
-        IdxRangeRTheta grid(get_idx_range<GridR, GridTheta>(density_host));
-        AdvectionFieldFinder advection_field_computer(m_mapping);
+        IdxRangeRTheta grid(get_idx_range(density_host));
 
-        host_t<PolarSplineMemRTheta> electrostatic_potential_coef(
+        // Data
+        DFieldMemRTheta electrical_potential_alloc(grid);
+        host_t<Spline2DMem> density_coef_host(get_spline_idx_range(m_builder));
+        host_t<PolarSplineMemRTheta> electrostatic_potential_coef_alloc_host(
                 ddc::discrete_space<PolarBSplinesRTheta>().full_domain());
 
-        host_t<DFieldMemRTheta> electrical_potential0_host(grid);
-        DFieldMemRTheta electrical_potential0(grid);
-        host_t<Spline2DMem> density_coef(get_spline_idx_range(m_builder));
-        m_builder(get_field(density_coef), get_const_field(density_host));
+        auto electrical_potential_alloc_host
+                = ddc::create_mirror_view(get_field(electrical_potential_alloc));
+        auto density_alloc = ddc::create_mirror_view(Kokkos::DefaultExecutionSpace(), density_host);
+
+        // Fields
+        DFieldRTheta electrical_potential = get_field(electrical_potential_alloc);
+        host_t<DFieldRTheta> electrical_potential_host = get_field(electrical_potential_alloc_host);
+        host_t<PolarSplineRTheta> electrostatic_potential_coef_host
+                = get_field(electrostatic_potential_coef_alloc_host);
+        DFieldRTheta density = get_field(density_alloc);
+
+        // Operators
+        AdvectionFieldFinder advection_field_computer(m_mapping);
         PoissonLikeRHSFunction const
-                charge_density_coord(get_const_field(density_coef), m_spline_evaluator);
-        m_poisson_solver(charge_density_coord, get_field(electrical_potential0));
-        ddc::parallel_deepcopy(electrical_potential0, electrical_potential0_host);
+                charge_density(get_const_field(density_coef_host), m_spline_evaluator);
+
+        // Setup
+        m_builder(get_field(density_coef_host), get_const_field(density_host));
```

Comment:
```suggestion
        m_builder(density_coef_host, get_const_field(density_host));
```
? 

or change for `density_coef_alloc_host`.

Reply:
[c246b3e](https://github.com/gyselax/gyselalibxx/pull/395/commits/c246b3e9a7accf6fda3a0a7e714d52383fdceaab)